### PR TITLE
Feature/admin request view

### PR DESCRIPTION
**Pull request by Hairax**
### Description

A view was created so that the administrator can see a list of trainer requests and can accept or deny them.

## Step 1
To generate the complete flow, we must first have a normal user and an admin user
For the admin user, go to the keycloak service, create a user and create the "Admin" role within the nextjs client, after this assign the role to the admin user
## Step 2
Log in with the normal user and create a request to become a trainer
## Step 3
Log in as administrator and go to the route " http://localhost:3000/AdminChoachRequest "
If you are logged in with an administrator account, it should show you the page, otherwise it should redirect you to the access denied page
## Step 4
Accept or Reject the request

With that, the flow ends, remember to test that it is responsive

### CODE REVIEW

File: apps/frontend/app/AdminChoachRequest/page.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  const [isClient, setIsClient] = useState(false);
+  const router = useRouter();
+
+  useEffect(() => {
+    setIsClient(true);
+  }, []);
+
+  useEffect(() => {
+    if (isClient && !isLoading && user && user.role !== 'admin') {
+      router.push('/access-denied');
```
Comment (GastonGutierrezC) [inline apps/frontend/app/AdminChoachRequest/page.tsx]: remember that we have a file where all the routes are, put it in there.

File: apps/frontend/app/AdminChoachRequest/hooks/useCoachRequestUpdate.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export const useCoachRequestUpdate = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const updateTrainerRequest = async (id: string, updatedData: Partial<TrainerRequest>) => {
+    setIsLoading(true);
+    setError(null);
+
+    try {
+      const response = await fetch(`http://localhost:4444/trainer-requests/${id}`, {
```
Comment (GastonGutierrezC) [inline apps/frontend/app/AdminChoachRequest/hooks/useCoachRequestUpdate.tsx]: in utils there is a file for connections and requests to the back end, place it there.

File: apps/frontend/app/AdminChoachRequest/hooks/useCoachRequestGet.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { useEffect, useState } from 'react';
+import { TrainerRequest } from "../../types";
+
+export const useCoachRequestGet = () => {
+    const [data, setData] = useState<TrainerRequest[]>([]);
+    
+    useEffect(() => {
+        async function fetchData() {
+        const response = await fetch('http://localhost:4444/trainer-requests/');
```
Comment (GastonGutierrezC) [inline apps/frontend/app/AdminChoachRequest/hooks/useCoachRequestGet.tsx]: in utils there is a file for connections and requests to the back end, place it there.

**Comment by GastonGutierrezC** [comment]
Screenshot from 2024-11-29 11-01-34
 place a label or something to be able to differentiate the trainers admitted, rejected or which ones still need to be reviewed.

**Comment by GastonGutierrezC** [comment]
Screenshot from 2024-11-29 10-56-53
instead of the user ID you could enter the user's name

File: apps/frontend/routes.js [hunk 4]
```diff
@@ -7,7 +7,9 @@ const RoutesNavigation = {
     Calendar: "/Calendar",
     ChatBot: "/ChatBot",
     CreateRoutine: "/routines/create",
-    YouRoutine: "/routines/yourRoutines"
+    YouRoutine: "/routines/yourRoutines",
+    AdminChoachRequest: "/AdminChoachRequest",
```
Comment (GastonGutierrezC) [inline apps/frontend/routes.js]: the route should at least start with lowercase letters

**Review by GastonGutierrezC — APPROVED** [review]

**Review by JheremyTancara — APPROVED** [review]

**Pull request merged**